gencred/cmd/gencred/main.go

```diff
@@ -170,25 +282,109 @@ func Main() {
 	var o options
 
 	o.parseFlags()
-	if err := o.validateFlags(); err != nil {
-		util.PrintErrAndExit(err)
-	}
-
-	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
-	configOverrides := &clientcmd.ConfigOverrides{CurrentContext: o.context}
-	kubeconfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, configOverrides)
-
-	config, err := kubeconfig.ClientConfig()
+	c, err := o.defaultAndValidateFlags()
 	if err != nil {
 		util.PrintErrAndExit(err)
 	}
 
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		util.PrintErrAndExit(err)
+	if o.refreshInterval == 0 {
+		if err := runOnce(*c); err != nil {
+			util.PrintErrAndExit(err)
+		}
+		return
 	}
 
-	if err = writeConfig(o, clientset); err != nil {
-		util.PrintErrAndExit(err)
+	defer interrupts.WaitForGracefulShutdown()
+	interrupts.Tick(func() {
+		runOnce(*c)
+	}, func() time.Duration { return o.refreshInterval })
+}
+
+func runOnce(c config) error {
+	// Make sure process everyone before crying.
+	var errs []error
+	var config *rest.Config
+	for _, cc := range c.Clusters {
+		if cc.GKEConnection != nil && cc.Context != nil {
+			errs = append(errs, errors.New("gke and context are mutually exclusive"))
+			continue
+		}
+		if cc.Duration.Duration == 0 {
+			cc.Duration = &metav1.Duration{Duration: defaultDuration}
+		}
```

Comment:
nit: Should we move these to `defaultAndValidateFlags` with the rest of the config validation?